src/sinks/aws_cloudwatch_logs/request.rs

```diff
@@ -70,33 +74,24 @@ impl CloudwatchFuture {
 }
 
 impl Future for CloudwatchFuture {
-    type Item = ();
-    type Error = CloudwatchError;
+    type Output = Result<(), CloudwatchError>;
 
-    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+    fn poll(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
+        let this = self.get_mut();
         loop {
-            match &mut self.state {
+            match &mut this.state {
```

Comment:
This is almost java 🤔 

You might be able to get away with the `deref` impl on `Pin` but this works just fine.